tests/gtest/radio_spinel_rcp_test.cpp

```diff
@@ -371,3 +371,126 @@ TEST(RadioSpinelTransmit, shouldSkipCsmaBackoffWhenCsmaCaIsEnabledAndMaxBackoffs
 
     platform.GoInMs(1000);
 }
+
+TEST(RadioSpinelSrcMatch, shouldBeAbleToEnableRadioSrcMatch)
+{
+    FakeCoprocessorPlatform platform;
+
+    platform.SrcMatchEnable(false);
+    ASSERT_EQ(platform.mRadioSpinel.Enable(FakePlatform::CurrentInstance()), kErrorNone);
+    ASSERT_EQ(platform.mRadioSpinel.EnableSrcMatch(true), kErrorNone);
+    ASSERT_EQ(platform.mSrcMatchEnabled, true);
+}
+
+TEST(RadioSpinelSrcMatch, shouldBeAbleToDisableRadioSrcMatch)
+{
+    FakeCoprocessorPlatform platform;
+
+    platform.SrcMatchEnable(true);
+    ASSERT_EQ(platform.mRadioSpinel.Enable(FakePlatform::CurrentInstance()), kErrorNone);
+    ASSERT_EQ(platform.mRadioSpinel.EnableSrcMatch(false), kErrorNone);
+    ASSERT_EQ(platform.mSrcMatchEnabled, false);
+}
+
+TEST(RadioSpinelSrcMatch, shouldBeAbleToAddRadioSrcMatchShortEntry)
+{
+    FakeCoprocessorPlatform platform;
+
+    platform.SrcMatchEnable(true);
+
+    constexpr uint16_t kTestShortAddr = 0x1234;
+
+    ASSERT_EQ(platform.mSrcMatchShortAddrs.count(kTestShortAddr), 0);
+
+    ASSERT_EQ(platform.mRadioSpinel.Enable(FakePlatform::CurrentInstance()), kErrorNone);
+    ASSERT_EQ(platform.mRadioSpinel.AddSrcMatchShortEntry(kTestShortAddr), kErrorNone);
+
+    ASSERT_EQ(platform.mSrcMatchShortAddrs.count(kTestShortAddr), 1);
+}
+
+TEST(RadioSpinelSrcMatch, shouldBeAbleToClearRadioSrcMatchShortEntry)
+{
+    constexpr uint16_t      kTestShortAddr = 0x1234;
+    FakeCoprocessorPlatform platform;
+
+    platform.SrcMatchEnable(true);
+    platform.SrcMatchAddShortEntry(kTestShortAddr);
+
+    ASSERT_EQ(platform.mSrcMatchShortAddrs.count(kTestShortAddr), 1);
+
+    ASSERT_EQ(platform.mRadioSpinel.Enable(FakePlatform::CurrentInstance()), kErrorNone);
+    ASSERT_EQ(platform.mRadioSpinel.ClearSrcMatchShortEntry(kTestShortAddr), kErrorNone);
+
+    ASSERT_EQ(platform.mSrcMatchShortAddrs.count(kTestShortAddr), 0);
+}
+
+TEST(RadioSpinelSrcMatch, shouldBeAbleToAddRadioSrcMatchExtEntry)
+{
+    FakeCoprocessorPlatform platform;
+
+    platform.SrcMatchEnable(true);
+    platform.mSrcMatchShortAddrs.clear();
+
+    constexpr otExtAddress kTestExtAddr{0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, 0x88};
+    constexpr otExtAddress kTestExtAddrReserved{0x88, 0x77, 0x66, 0x55, 0x44, 0x33, 0x22, 0x11};
```

Comment:
```suggestion
    constexpr otExtAddress kTestExtAddrReserved{0x88, 0x77, 0x66, 0x55, 0x44, 0x33, 0x22, 0x11};
```
Do you mean `kTestExtAddrReversed`?